runtime/curl.h

```diff
@@ -4,18 +4,61 @@
 
 #pragma once
 
-#include "common/smart_ptrs/singleton.h"
+#include <functional>
 
+#include "common/smart_ptrs/singleton.h"
+ 
 #include "runtime/kphp_core.h"
+#include "runtime/critical_section.h"
+#include "runtime/streams.h"
 
 using curl_easy = int64_t;
 
+namespace curl {
+  using on_read_callable = std::function<string(curl_easy ch, Stream stream, size_t length)>;
```

Comment:
## Aliases for the callable arguments of the curl_setopt() function:
Added to the runtime:
- on_read_callable for [CURLOPT_READFUNCTION](https://curl.se/libcurl/c/CURLOPT_READFUNCTION.html)
- on_write_callable for [CURLOPT_WRITEFUNCTION](https://curl.se/libcurl/c/CURLOPT_WRITEFUNCTION.html)
- on_progress_callable for [CURLOPT_PROGRESSFUNCTION](https://curl.se/libcurl/c/CURLOPT_PROGRESSFUNCTION.html)
- on_xferinfo_callable for [CURLOPT_XFERINFOFUNCTION](https://curl.se/libcurl/c/CURLOPT_XFERINFOFUNCTION.html)